lib/dat-worker-pool.rb

```diff
@@ -77,14 +69,21 @@ def worker_available?
     @workers_waiting.count > 0
   end
 
-  def on_worker_error(&block);    @on_worker_error_callbacks    << block; end
-  def on_worker_start(&block);    @on_worker_start_callbacks    << block; end
-  def on_worker_shutdown(&block); @on_worker_shutdown_callbacks << block; end
-  def on_worker_sleep(&block);    @on_worker_sleep_callbacks    << block; end
-  def on_worker_wakeup(&block);   @on_worker_wakeup_callbacks   << block; end
+  def on_worker_start_callbacks;    @worker_class.on_start_callbacks;    end
+  def on_worker_shutdown_callbacks; @worker_class.on_shutdown_callbacks; end
+  def on_worker_sleep_callbacks;    @worker_class.on_sleep_callbacks;    end
+  def on_worker_wakeup_callbacks;   @worker_class.on_wakeup_callbacks;   end
+  def on_worker_error_callbacks;    @worker_class.on_error_callbacks;    end
+  def before_work_callbacks;        @worker_class.before_work_callbacks; end
+  def after_work_callbacks;         @worker_class.after_work_callbacks;  end
 
-  def before_work(&block); @before_work_callbacks << block; end
-  def after_work(&block);  @after_work_callbacks  << block; end
+  def on_worker_start(&block);    @worker_class.on_start(&block);    end
+  def on_worker_shutdown(&block); @worker_class.on_shutdown(&block); end
+  def on_worker_sleep(&block);    @worker_class.on_sleep(&block);    end
+  def on_worker_wakeup(&block);   @worker_class.on_wakeup(&block);   end
+  def on_worker_error(&block);    @worker_class.on_error(&block);    end
+  def before_work(&block);        @worker_class.before_work(&block); end
+  def after_work(&block);         @worker_class.after_work(&block);  end
```

Comment:
@jcredding do we need to even do this anymore?  You didn't add the prepend demeters here and I was wondering why we need this at all?  Just backwards compatibility for now?